site/app/controllers/admin/AdminGradeableController.php

```diff
@@ -1083,16 +1102,71 @@ private function createGradeable($gradeable_id, $details) {
         if ($tonight->diff($this->core->getDateTimeNow())->h < 12) {
             $tonight->add(new \DateInterval('P1D'));
         }
+
+        $ta_view_start_date = (
+            isset($details['ta_view_start_date'])
+                ? new \DateTime($details['ta_view_start_date'], $this->core->getConfig()->getTimezone())
+                : (clone $tonight)
+        );
```

Comment:
Very similar logic is being repeated here.  Would it be better to simplify it, either by creating a function to do it, or by setting array key-value pairs of an array in a loop?  Regarding the actual construction of new DateTime objects, I wonder if it would make sense to create a method in the Core library class to do this, similar to the `getDateTimeNow()` method?

https://github.com/Submitty/Submitty/blob/b289ce22f17b429fc8f8e32761e97b60981404ec/site/app/libraries/Core.php#L769-L771